pkg/controller/installation/products/ups/reconciler_test.go

```diff
@@ -0,0 +1,221 @@
+	"github.com/integr8ly/integreatly-operator/pkg/controller/installation/products/config"
+
+	moqclient "github.com/integr8ly/integreatly-operator/pkg/client"
+	"github.com/pkg/errors"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func basicConfigMock() *config.ConfigReadWriterMock {
+	return &config.ConfigReadWriterMock{
+		ReadUpsFunc: func() (ups *config.Ups, e error) {
+			return config.NewUps(config.ProductConfig{}), nil
+		},
+		WriteConfigFunc: func(config config.ConfigReadable) error {
+			return nil
+		},
+	}
+}
+
+func errorConfigMock() *config.ConfigReadWriterMock {
+	return &config.ConfigReadWriterMock{
+		ReadUpsFunc: func() (ups *config.Ups, e error) {
+			return config.NewUps(config.ProductConfig{}), nil
+		},
+		WriteConfigFunc: func(config config.ConfigReadable) error {
+			return errors.New("some error")
+		},
+	}
+}
+
+func basicRouteMock() *v1Route.Route {
+	return &v1Route.Route{
+		TypeMeta: v1.TypeMeta{
+			Kind:       "Route",
+			APIVersion: "v1",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Namespace: "ups",
+			Name:      defaultRoutename,
+		},
+	}
+}
+
+func TestReconciler_ReconcileCustomResource(t *testing.T) {
+	scheme := runtime.NewScheme()
+	upsv1alpha1.SchemeBuilder.AddToScheme(scheme)
+
+	cases := []struct {
+		Name           string
+		client         client.Client
+		FakeConfig     *config.ConfigReadWriterMock
+		Installation   *v1alpha1.Installation
+		ExpectErr      bool
+		ExpectedStatus v1alpha1.StatusPhase
+		FakeMPM        *marketplace.MarketplaceInterfaceMock
+	}{
+		{
+			Name:           "UPS Test: test custom resource is reconciled and phase complete returned",
+			ExpectedStatus: v1alpha1.PhaseCompleted,
+			FakeMPM:        &marketplace.MarketplaceInterfaceMock{},
+			Installation:   &v1alpha1.Installation{},
+			FakeConfig:     basicConfigMock(),
+			client: fake.NewFakeClientWithScheme(scheme,
+				&upsv1alpha1.UnifiedPushServer{
+					ObjectMeta: v1.ObjectMeta{
+						Namespace: "ups",
+						Name:      "ups",
+					},
+					Status: upsv1alpha1.UnifiedPushServerStatus{
+						Phase: upsv1alpha1.PhaseComplete,
+					},
+				}),
+		},
+		{
+			Name:           "UPS Test: Phase failed when error in creating custom resource",
+			ExpectedStatus: v1alpha1.PhaseFailed,
+			FakeMPM:        &marketplace.MarketplaceInterfaceMock{},
+			Installation:   &v1alpha1.Installation{},
+			FakeConfig:     basicConfigMock(),
+			client: &moqclient.SigsClientInterfaceMock{
+				GetFunc: func(ctx context.Context, key types.NamespacedName, obj runtime.Object) error {
+					return k8serr.NewNotFound(schema.GroupResource{Group: upsv1alpha1.SchemeGroupVersion.Group, Resource: "UnifiedPushServer"}, "Not found")
```

Comment:
It might be fine just to do
```suggestion
return k8serr.NewNotFound(schema.GroupResource{}, "unifiedpushserver")
```